js/donations-visualiser.js

```diff
@@ -125,6 +125,27 @@ d3.select("#receipt-type-select-all").on("click", function() { selectAll('#recei
 d3.select("#receipt-type-select-clear").on("click", function() { clearSelection('#receipt_type_select'); });
 d3.select("#clear-search").on("click", clearSearch);
 
+// on 'escape' key press, close the info window, and zoomToFit
+window.addEventListener("keydown", function (event) {
+  if (event.defaultPrevented) return;
+
+  var handled = false;
+  if (event.key === 27) {
+    handled = true;
+  } else if (event.keyIdentifier === 27) {
+    handled = true;
+  } else if (event.keyCode === 27) {
+    handled = true;
+  }
```

Comment:
yep, no worries.

On 8 June 2016 at 22:53, Tim Osborn <notifications@github.com> wrote:

> In js/donations-visualiser.js
> <https://github.com/river-jade/donations-visualiser/pull/16#discussion_r66248926>
> :
>
> > @@ -125,6 +125,27 @@ d3.select("#receipt-type-select-all").on("click", function() { selectAll('#recei
> >  d3.select("#receipt-type-select-clear").on("click", function() { clearSelection('#receipt_type_select'); });
> >  d3.select("#clear-search").on("click", clearSearch);
> >
> > +// on 'escape' key press, close the info window, and zoomToFit
> > +window.addEventListener("keydown", function (event) {
> > +  if (event.defaultPrevented) return;
> > +
> > +  var handled = false;
> > +  if (event.key === 27) {
> > +    handled = true;
> > +  } else if (event.keyIdentifier === 27) {
> > +    handled = true;
> > +  } else if (event.keyCode === 27) {
> > +    handled = true;
> > +  }
>
> this was just rushed, really - it is cross browser, copypasta direct from
> mdn... prob could be replaced with some much shorter jquery, i'd say
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/river-jade/donations-visualiser/pull/16/files/2a49d5a712e0d26812b77159fe82ef7ef30524fe..1ba2e76cc5fc7649a9bd51e5077e878f8947d34b#r66248926>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AArMkChLwDmusSLdsjYT5Is9JLp0Mharks5qJrs2gaJpZM4Ivg33>
> .
>